services/app-web/src/gqlHelpers/fetchHealthPlanPackageWrapper.ts

```diff
@@ -0,0 +1,191 @@
+import { ApolloError } from '@apollo/client'
+import {
+    useFetchHealthPlanPackageQuery,
+    FetchHealthPlanPackageQuery,
+    HealthPlanRevision,
+    HealthPlanRevisionEdge,
+    HealthPlanPackage,
+} from '../gen/gqlClient'
+import { HealthPlanFormDataType } from '../common-code/healthPlanFormDataType'
+import { base64ToDomain } from '../common-code/proto/healthPlanFormDataProto'
+import { recordJSException } from '../otelHelpers'
+
+type FetchLoading = {
+    status: 'LOADING'
+}
+
+type FetchError = {
+    status: 'ERROR'
+    error: Error
+}
+
+type FetchSuccess = {
+    status: 'SUCCESS'
+    data: FetchHealthPlanPackageQuery
+}
+
+type FetchHealthPlanPackageResult = FetchLoading | FetchError | FetchSuccess
+
+// These apollo results are not actually supposed to be used simultaneously, so we can make a result type that only returns the correct one
+function wrapApolloResult({
+    loading,
+    error,
+    data,
+}: {
+    loading: boolean
+    error?: ApolloError
+    data: FetchHealthPlanPackageQuery | undefined
+}): FetchHealthPlanPackageResult {
+    if (loading) {
+        return {
+            status: 'LOADING',
+        }
+    }
+
+    if (error) {
+        return {
```

Comment:
🔴 issue: We are missing out on uniform OTEL logging here. There are several types of errors that can happen that can log for before returning things to the component. This is what the  new`handleApolloError` function is intended for.

Reply:
Great callout, I actually ended pulling all the otel error logging out of the wrapper and instead am explicitly handling the errors from inside the pages themselves. I think that trying to do as much of that error-logging as possible from those pages directly instead of inside library code is going to make thing easiest. In the page itself is where we have all the context neccecary to log errors correctly, an easy example of that is that we can pass isAuthenticated to handleApolloError with confidence. I’d love to chat about this at our next eng meeting. 

Another way this makes sense to me is that the wrapper returns the error, and our pages that call this have to respond to that error, I think it makes the most sense to assume as a page that you need to fully handle logging the error rather than assuming that that has been done.

Reply:
Sure! I'm fine with that boundary, just having it consistent and using the apollo error helper so we get robust errors is my interest.